pkg/printers/printer_test.go

```diff
@@ -0,0 +1,212 @@
+package printers
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/logutils"
+	"github.com/golangci/golangci-lint/pkg/report"
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+func unmarshalFile(t *testing.T, filename string, v any) {
+	t.Helper()
+
+	file, err := os.ReadFile(filepath.Join("fixtures", filename))
```

Comment:
Usually, helper files for tests are placed under the `testdata` directory.

Reply:
`testdata` is for (Go) files that are ignored by the compiler.

For me, it's not the best place for basic fixture files.